bot-components/GitHub_queries.mli

```diff
@@ -0,0 +1,104 @@
+val send_graphql_query :
+     bot_info:Utils.bot_info
+  -> < parse: Yojson.Basic.t -> 'a
+     ; query: string
+     ; variables: Yojson.Basic.t
+     ; .. >
+  -> ('a, string) result Lwt.t
+
+type backport_info =
+  {backport_to: string; request_inclusion_column: int; backported_column: int}
+
+type full_backport_info =
+  {backport_info: backport_info list; rejected_milestone: string}
+
+val get_backport_info :
+  bot_info:Utils.bot_info -> string -> full_backport_info option
+
+type project_card =
+  { id: string
+  ; column: GitHub_GraphQL.project_column option
+  ; columns: GitHub_GraphQL.project_column list }
+
+val pull_request_milestone_and_cards :
+     bot_info:Utils.bot_info
+  -> owner:string
+  -> repo:string
+  -> number:int
+  -> (project_card list * GitHub_GraphQL.milestone option, string) result Lwt.t
+
+type mv_card_to_column_input = {card_id: string; column_id: string}
+
+val backported_pr_info :
+     bot_info:Utils.bot_info
+  -> int
+  -> string
+  -> mv_card_to_column_input option Lwt.t
+
+val get_pull_request_id_and_milestone :
+     bot_info:Utils.bot_info
+  -> owner:string
+  -> repo:string
+  -> number:int
+  -> ((string * int * full_backport_info) option, string) result Lwt.t
+
+val get_team_membership :
+     bot_info:Utils.bot_info
+  -> org:string
+  -> team:string
+  -> user:string
+  -> (bool, string) result Lwt.t
+
+val get_pull_request_refs :
+     bot_info:Utils.bot_info
+  -> owner:string
+  -> repo:string
+  -> number:int
+  -> (string GitHub_subscriptions.pull_request_info, string) result Lwt.t
```

Comment:
Instead of having some types defined in `GitHub_subscriptions`, some defined in `GitHub_queries` and some in `GitHub_mutations`, we could consider introducing a new `GitHub_types` module to gather them. It would also present the advantage of not requiring to duplicate the type definition with a `modules_without_implementation`. See https://dune.readthedocs.io/en/stable/dune-files.html#library